MapleServer2/PacketHandlers/Game/InsigniaHandler.cs

```diff
@@ -14,15 +17,80 @@ public class InsigniaHandler : GamePacketHandler
 
         public override void Handle(GameSession session, PacketReader packet)
         {
+
+            ProcessInsignias(session);
+
             short insigniaId = packet.ReadShort();
 
             if (insigniaId < 0)
             {
                 return;
             }
 
-            session.Player.InsigniaId = insigniaId;
-            session.FieldManager.BroadcastPacket(InsigniaPacket.UpdateInsignia(session, insigniaId, true));
+            if (session.Player.AvaliableInsignias.Contains(insigniaId))
+            {
+                session.Player.InsigniaId = insigniaId;
+            }
+            else
+            {
+                insigniaId = 0;
+            }
+
+            session.FieldManager.BroadcastPacket(InsigniaPacket.UpdateInsignia(session, insigniaId, insigniaId != 0));
+        }
+
+        private void ProcessInsignias(GameSession session)
```

Comment:
Do we know if the checks and such here are not included in any xml file that can be used?

Reply:
I will look!